app/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -178,15 +178,28 @@ internal constructor(
   }
 
   /**
-   * Returns a flow of [MapCardUiData] associated with the active survey's LOIs and adhoc jobs for
-   * displaying the cards.
+   * Returns a flow of [DataCollectionEntryPointData] associated with the active survey's LOIs and
+   * adhoc jobs for displaying the cards.
    */
-  fun getMapCardUiData(): Flow<Pair<List<MapCardUiData>, Int>> =
-    loisInViewport.combine(adHocLoiJobs) { lois, jobs ->
-      val loiCards = lois.map { MapCardUiData.LoiCardUiData(it) }
-      val jobCards = jobs.map { MapCardUiData.AddLoiCardUiData(it) }
-
-      Pair(loiCards + jobCards, lois.size)
+  fun getDataCollectionEntryPoints():
+    Flow<
+      Pair<
+        DataCollectionEntryPointData.SelectedLoiSheetData?,
+        List<DataCollectionEntryPointData.AdHocDataCollectionButtonData>,
+      >
+    > =
+    combine(loisInViewport, featureClicked, adHocLoiJobs) { loisInView, feature, jobs ->
+      val loiCard =
+        loisInView
+          .filter { it.geometry == feature?.geometry }
+          .firstOrNull()
+          ?.let { DataCollectionEntryPointData.SelectedLoiSheetData(it) }
+      if (loiCard == null && feature != null) {
+        // The feature is not in view anymore.
+        featureClicked.value = null
```

Comment:
It feels like the resetting logic should live somewhere else. This method should only return the data and shouldn't do anything other than that.

Reply:
I changed the name to reflect the true `processDataCollectionEntryPoints`, I kind of wanted a single coroutine handling all the different states for the job cards.